segmentation/utils/spatialorgmarkerexpression_utils.py

```diff
@@ -0,0 +1,263 @@
+
+
+def calculate_channel_spatial_enrichment(dist_matrix, marker_thresholds, cell_array,
+                                         excluded_colnames=["cell_size", "Background", "HH3",
+                                                            "summed_channel", "label", "area",
+                                                            "eccentricity", "major_axis_length", "minor_axis_length",
+                                                            "perimeter", "fov"],
+                                         patient_idx=30):
+    """Spatial enrichment analysis to find significant interactions between cells
+        expressing different markers.
+        Uses bootstrapping to permute cell labels randomly.
+                Args
+                    dist_matrix: cells x cells matrix with the euclidian
+                        distance between centers of corresponding cells
+                    marker_thresholds: threshold
+                        values for positive marker expression
+                    cell_array: data including points, cell labels, and
+                        cell expression matrix for all markers
+                Returns
+                    close_num: marker x marker matrix with counts for cells
+                        positive for corresponding markers
+                    close_num_rand: random positive marker counts
+                        for every permutation in the bootstrap
+                    z: z scores for corresponding markers
+                    muhat: predicted mean values
+                        of close_num_rand random distribution
+                    sigmahat: predicted standard deviation values of close_num_rand
+                        random distribution
+                    p: p values for corresponding markers, for both positive
+                        and negative enrichment
+                    h: matrix indicating whether
+                        corresponding marker interactions are significant
+                    adj_p: fdh_br adjusted p values
+                    marker_titles: list of markers"""
+
+    # Setup input and parameters
+    #point = 6
+    point = "Point8"
+    dist_lim = 100
+    # column in expression matrix with patient labels
+    # patient_idx = 0
+
+    # subsetting expression matrix
+    data_all = cell_array
+    # identifies columns with markers
+    # marker_inds = [7, 8]  # + list(range(10, 44))
+    # subsets the expression matrix to only have marker columns
+    # data_markers = data_all.loc[:, data_all.columns[marker_inds]]
+    data_markers = data_all.drop(excluded_colnames, axis=1)
+    # list of all markers
+    # marker_titles = data_all.columns[marker_inds]
+    marker_titles = data_markers.columns
+    # length of marker list
+    marker_num = len(marker_titles)
+
+    # subsetting threshold matrix to only include column with threshold values
+    # thresh_vec = marker_thresholds.iloc[1:38, 1]
+    thresh_vec = marker_thresholds.iloc[0:20, 1]
```

Comment:
This index should be determined by the data. If you set it to be 0:20, it's not going to work if there are different values.